File: app/src/main/java/com/example/android/todolist/addtaskactivity/AddTaskViewModel.java
package com.example.android.todolist.addtaskactivity;

import android.arch.lifecycle.LiveData;
import android.arch.lifecycle.ViewModel;

import com.example.android.todolist.database.AppDatabase;
import com.example.android.todolist.database.TaskEntry;

/**
 * Created by raed on 9/20/18.
 */

public class AddTaskViewModel extends ViewModel {

    private LiveData<TaskEntry> task;

    public AddTaskViewModel (AppDatabase database, int taskId) {
        task = database.taskDao().loadTaskById(taskId);
    }

    public LiveData<TaskEntry> getTask () {
        return task;
    }
}

File: app/src/main/java/com/example/android/todolist/addtaskactivity/AddTaskViewModel.kt
package com.example.android.todolist.addtaskactivity

import android.arch.lifecycle.LiveData
import android.arch.lifecycle.ViewModel
import com.example.android.todolist.database.AppDatabase
import com.example.android.todolist.database.TaskEntry

/**
 * Created by Raed Saeed on 8/3/2019.
 */
class AddTaskViewModel(appDatabase: AppDatabase, taskId: Int) : ViewModel() {
    val task: LiveData<TaskEntry> = appDatabase.taskDao().loadTaskById(taskId)
}